ouroboros-consensus-test/src/Test/Ouroboros/Consensus/ChainGenerator/Counting.hs

```diff
@@ -0,0 +1,336 @@
+{-# LANGUAGE ExistentialQuantification #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE PolyKinds #-}
+{-# LANGUAGE Rank2Types #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE ViewPatterns #-}
+
+-- | Very strong types for working with indices, counts, etc within sequences.
+module Test.Ouroboros.Consensus.ChainGenerator.Counting (
+  -- * general counts
+  Count (Count),
+  (-),
+  (+),
+  getCount,
+  forgetBase,
+  forgetElem,
+  -- * indices and sizes
+  Preds,
+  Index,
+  Size,
+  Total,
+  forRangeM_,
+  lastIndex,
+  range, 
+  uniformIndex,
+  -- * windows
+  Lbl (Lbl),
+  SomeWindow (SomeWindow),
+  Win,
+  Contains (Contains, UnsafeContains),
```

Comment:
I  intentionally export `Unsafe*` in general; it's the user's fault for using it without understanding it.

I think it's worse to make it more difficult for someone to explore some additional behavior that would otherwise necessitate patching our code.